### ISSUE TITLE
CUDA intrinsics outside of kernel (in header) will lead to launcher compilation problem

### ISSUE THREAD

**Issue by ypanke** [libocca/occa-transpiler#180]
Code:
```
void myfn() {
    float var = 10.0;
    float res = __exp10f(var);
}

@kernel void hello_kern() {
    for (int i = 0; i < 10; ++i; @outer) {
        for (int j = 0; j < 10; ++j; @inner) {
        }
    }
}
```

Will generate launcher
```
#include <occa/core/kernel.hpp>

void myfn() {
  float var = 10.0;
  float res = __exp10f(var);
}

extern "C" void hello_kern(occa::modeKernel_t **deviceKernels) {
  {
    occa::dim outer, inner;
    outer.dims = 1;
    inner.dims = 1;
    int i = 0;
    outer[0] = (10) - (0);
    int j = 0;
    inner[0] = (10) - (0);
    occa::kernel kernel(deviceKernels[0]);
    kernel.setRunDims(outer, inner);
    kernel();
  };
}
```

leaving the `__exp10f` call

**Comment by kchabSS** [comment]
Does the original OCCA generate the same result?

**Comment by IuriiKobein** [comment]
Fixed in #220
